bebraven/boomerang
# Initial configuration of Boomerang for Braven.

- Disables a number of plugins we're not using out of the gate.
- Adds state and trace fields to every beacon so the server side
  can proxy the info to Honeycomb
- Initializes Boomerang with settings that generate pretty nice traces
  when used with this platform PR: https://github.com/bebraven/platform/pull/331
  for the initial release of this

### Test Plan:
- upload `build/boomerang-1.0.0.js` to our `boomerangjs` S3 bucket
- load a Project in Canvas (published as an LTI assignment) that has
  it's xAPI statements sent from JS using the honeycomb.js wrapper in
  the platform code in order to add instrumentation
- Query for the trace.traceid in the Honeycomb UI and look at the trace.
  It should show the `javascript.page.load` span and the `populatePreviousAnswers`
  span. The duration of the span should match what is expected. The waterfall view
  should tie it all into a trace for the original Project controller.
- Force the xAPI JS to throw an error when sending an xAPI statement by commenting
  out the activity_id of the payload. Look at the trace and notice the error fields
  in there, including the stack_trace.
